modules/core/src/test/scala/org/scalasteward/core/edit/RewriteTest.scala

```diff
@@ -840,6 +840,31 @@ class RewriteTest extends FunSuite {
     runApplyUpdate(update, original, expected)
   }
 
+  test("issue-2877: sbt using same version in a val and a literal".fail) {
+    val update = ("org.scalatest".g % "scalatest".a % "3.2.13" %> "3.2.14").single
+    val original = Map(
+      "build.sbt" ->
+        """
+          |val ScalaTestVersion = "3.2.13"
+          |libraryDependencies ++= Seq(
+          |  "org.scalatest" %% "scalatest" % ScalaTestVersion,
+          |  "org.scalatest" %% "scalactic" % "3.2.13"
+          |)
+          |""".stripMargin
+    )
+    val expected = Map(
+      "build.sbt" ->
+        """
+          |val ScalaTestVersion = "3.2.14"
+          |libraryDependencies ++= Seq(
+          |  "org.scalatest" %% "scalatest" % ScalaTestVersion,
+          |  "org.scalatest" %% "scalactic" % "3.2.14"
```

Comment:
Why should the scalactic version be bumped if the update only contains scalatest? I'd say it is correct to not bump the scalactic version in this case and Scala Steward is already doing the right thing.

Also this looks like Scala Steward bumped the version of the `val` but you mentioned in #2877 that it is unable to so. Something is missing here.

Reply:
You're right, I messed up my own example while translating it to code :)

I pushed up a fix in  https://github.com/scala-steward-org/scala-steward/pull/2884/commits/102a8e47de8de36c70c8ce6c1622bfaf53dc6808.

I'll update the PR description to match that fix.

Reply:
That looks much better, thanks! :-) My guess is that the comma in `ScalaTestVersion,` is tripping Scala Steward up. You could try removing it and see if that fixes the test. If you want to dig into the code, I think that [`scalaValInDependencyDefPositions`](https://github.com/scala-steward-org/scala-steward/blob/12362b512ad12bab690f2d12554595bc84ef1166/modules/core/src/main/scala/org/scalasteward/core/edit/update/Selector.scala#L46) should have returned a `VersionPosition`  of `val ScalaTestVersion = "3.2.13"` but it returned an empty list instead.

Reply:
Btw, if it is the comma that is causing trouble, the issue can be fixed by removing trailing commas in https://github.com/scala-steward-org/scala-steward/blob/89c66e36a42425857250b17be0b1be55aa0fad02/modules/core/src/main/scala/org/scalasteward/core/edit/update/data/ModulePosition.scala#L31-L32

Reply:
Gotcha:
```Scala
  def unwrappedVersion: String =
    version.value.dropWhile(Set('"', '$', '{')).reverse.dropWhile(Set('"', '}', ',')).reverse
```
Fixed the issue 👍

Reply:
Ah, yep, it's the comma. I added a test case that uses 2x `libraryDependencies +=` instead of 1x `libraryDependencies ++=`, and the 2x case passes: https://github.com/scala-steward-org/scala-steward/pull/2884/commits/102a8e47de8de36c70c8ce6c1622bfaf53dc6808